2_adventures/001/lenn4rd/03_raindrops_exercise/raindrops.rb

```diff
@@ -0,0 +1,25 @@
+class Fixnum
+  def prime_factor_of?(number)
+    number % self == 0
+  end
+end
+
+class Raindrops
+  PRIME_FACTORS = [3, 5, 7]
+
+  SOUNDS_FOR_FACTORS = {
+    '3' => 'Pling',
+    '5' => 'Plang',
+    '7' => 'Plong'
+  }
+  
+  def self.convert(number)
+    PRIME_FACTORS.each_with_object(sound = '') do |factor|
+      sound << SOUNDS_FOR_FACTORS[factor.to_s] if factor.prime_factor_of?(number)
+    end
+    
+    return number.to_s if sound.empty?
+    
+    sound
```

Comment:
You are quite right.  The second example will work because of the fact that both sides of the && have to be true... and if not, then the right hand side of the  || statement is used.  So if the sound is empty *and* if number evaluates as 'truthy' then that is used, and the || right hand side is not even evaluated.  If the sound.empty? is false, then the right hand side of the && is not used, it is already a false condition, and the || is automatically evaluated.

Soon you will see it and you will recognize it and immediately understand it. Perhaps even faster than the ternary, since the colon is hard visually to see and the question mark especially after a question mark is easy to miss.  Until then, you are right not to use it.

My code often toggles between the ternary and the pure logic statement.  If you are watching my contribution for tomorrow you will notice that.  Hopefully you will give me some feedback on it.

I'll see you on [exercism.io](http://exercism.io)!